### PR TITLE
Added advance filter to vcffilter

### PR DESCRIPTION
**Pull request by ffinfo**

### CODE REVIEW

File: biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala [hunk 1]
```diff
@@ -61,7 +61,8 @@ object VcfFilter extends ToolCommand {
                   diffGenotype: List[(String, String)] = Nil,
                   filterHetVarToHomVar: List[(String, String)] = Nil,
                   iDset: Set[String] = Set(),
-                  minGenomeQuality: Int = 0)
+                  minGenomeQuality: Int = 0,
+                  advanceGroups: List[List[String]] = Nil)
```
Comment (rhpvorderman) [inline biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala]: advancedGroups

Comment (rhpvorderman) [inline biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala]: advance**d**Groups

File: biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala [hunk 2]
```diff
@@ -153,6 +154,9 @@ object VcfFilter extends ToolCommand {
     opt[Int]("minGenomeQuality") unbounded () action { (x, c) =>
       c.copy(minGenomeQuality = x)
     }
+    opt[String]("advanceGroups") unbounded () action { (x, c) =>
```
Comment (rhpvorderman) [inline biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala]: Also the strings need to be refactored

File: biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala [hunk 3]
```diff
@@ -205,6 +209,7 @@ object VcfFilter extends ToolCommand {
           denovoTrio(record, cmdArgs.trioLossOfHet, onlyLossHet = true) &&
           resToDom(record, cmdArgs.resToDom) &&
           trioCompound(record, cmdArgs.trioCompound) &&
+          advanceGroupFilter(record, cmdArgs.advanceGroups) &&
```
Comment (rhpvorderman) [inline biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala]: advanced

File: biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala [hunk 4]
```diff
@@ -444,4 +449,24 @@ object VcfFilter extends ToolCommand {
   def inIdSet(record: VariantContext, idSet: Set[String]): Boolean = {
     record.getID.split(",").exists(idSet.contains)
   }
+
+  /**
+    * returns true when for all groups all or none members have a variants,
+    * records with partial groups are discarded
+    */
+  def advanceGroupFilter(record: VariantContext, groups: List[List[String]]): Boolean = {
```
Comment (rhpvorderman) [inline biopet-tools/src/main/scala/nl/lumc/sasc/biopet/tools/VcfFilter.scala]: idem

File: biopet-tools/src/test/scala/nl/lumc/sasc/biopet/tools/VcfFilterTest.scala [hunk 5]
```diff
@@ -261,4 +261,17 @@ class VcfFilterTest extends TestNGSuite with MockitoSugar with Matchers {
     inIdSet(record, Set("dummy")) shouldBe false
   }
 
+  @Test def testAdvanceGroup(): Unit = {
```
Comment (rhpvorderman) [inline biopet-tools/src/test/scala/nl/lumc/sasc/biopet/tools/VcfFilterTest.scala]: idem

**Review by rhpvorderman — CHANGES_REQUESTED** [review]

**Comment by ffinfo** [comment]
@rhpvorderman: should be all fixed now

**Pull request merged**